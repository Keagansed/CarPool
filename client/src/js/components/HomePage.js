import React, { Component } from 'react';
import { observer } from "mobx-react";

import { getFromStorage } from '../utils/localStorage.js'
import Carpools from './homepage/carpoolPage/Carpools'
import Navbar from './Navbar'
import NavTabs from './homepage/NavTabs'
import Routes from './homepage/routePage/Routes'
import Trips from './homepage/tripsPage/Trips'

@observer class HomePage extends Component{

    constructor(props){
        super(props);
        //========= Properties ===========
        this.state ={
          token:'',
        }; 
    }
    //========= Fetch Session Token ===========
    componentWillMount(){
        const obj = getFromStorage('sessionKey');
        if(obj && obj.token){
            //verify token
            const { token } = obj;
            fetch('/api/account/verify?token='+token)
            .then(res => res.json())
            .then(json => {
                if(json.success){
                    this.props.store.setToken(token);
                    this.props.store.setLoggedIn(true);
                }
            })
        }
    }

    render(){
        const { token } = this.props.store;

        return(
            <div className="HomePage">
                <div className="p-0 page-content">
                    <div className="container-fluid m-0">
                        <div className="row">
                            <div className="col-md-12 p-0">
                                <NavTabs />
                                <div className="tab-content">
                                    
                                    <Routes/>

                                    <Carpools />

                                    <Trips />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                
                <Navbar token={token}/>
            
            </div>
        );
    }
}

export default HomePage;