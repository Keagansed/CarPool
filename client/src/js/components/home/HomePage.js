import React, { Component } from 'react';
import { observer } from "mobx-react";

import Carpools from './carpoolPage/Carpools';
import { getFromStorage } from './../../utils/localStorage.js';
import Navbar from './../navbar/Navbar';
import NavTabs from './NavTabs';
import Routes from './routePage/Routes';
import RoutesStore from './../../stores/RoutesStore';
import Trips from './tripsPage/Trips';

@observer class HomePage extends Component{

    constructor(){
        super()

        this.state = {
            loading: true,
        }
    }

    //========= Fetch Session Token ===========
    componentWillMount(){
        const obj = getFromStorage('sessionKey');
        if(obj && obj.token){
            const { token } = obj;
            fetch('/api/account/verify?token='+token)
            .then(res => res.json())
            .then(json => {
                if(json.success){
                    this.props.store.token = token;

                    this.setState({
                        loading: false,
                    })
                }
            })
        }  
    }

    setTab = () => {
        const { store } = this.props;

        if(!this.state.loading)
        {
            if(store.routeTab === true)
            {
                return <Routes store={RoutesStore} token={this.props.store.token}/>;            
            }
            else if(store.carpoolTab === true)
            {
                return <Carpools/>;
            }
            else if(store.tripTab === true)
            {
                return <Trips/>;
            }
        }

    }

    render(){
        const { token } = this.props.store;
        
        return(
            <div className="size-100 bg-purple">
                    <NavTabs store={this.props.store} token={token}/>
                    {this.setTab()}
                    <Navbar token={token}/>
            </div>
        );
    }
}

export default HomePage;