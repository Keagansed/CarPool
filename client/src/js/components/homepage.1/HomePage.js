import React, { Component } from 'react';
import { observer } from "mobx-react";

import Carpools from './carpoolPage/Carpools';
import { getFromStorage } from '../utils/localStorage.js';
import Navbar from './Navbar';
import NavTabs from './NavTabs';
import Routes from './routePage/Routes';
import RoutesStore from './../../stores/RoutesStore';
import Trips from './homepage/tripsPage/Trips';

@observer class HomePage extends Component{

    constructor(){
        super()

        this.state = {
            loading: true,
        }
    }

    //========= Fetch Session Token ===========
    componentWillMount(){
        const obj = getFromStorage('sessionKey');
        if(obj && obj.token){
            const { token } = obj;
            fetch('/api/account/verify?token='+token)
            .then(res => res.json())
            .then(json => {
                if(json.success){
                    this.props.store.token = token;

                    this.setState({
                        loading: false,
                    })
                }
            })
        }  
    }

    setTab = () => {
        const { store } = this.props;

        if(!this.state.loading)
        {
            if(store.routeTab === true)
            {
                return <Routes store={RoutesStore} token={this.props.store.token}/>;            
            }
            else if(store.carpoolTab === true)
            {
                return <Carpools/>;
            }
            else if(store.tripTab === true)
            {
                return <Trips/>;
            }
        }

    }

    render(){
        const { token } = this.props.store;
        
        return(
            <div className="HomePage">
                <div className="p-0 page-content">
                    <div className="container-fluid m-0">
                        <div className="row">
                            <div className="col-md-12 p-0">
                                <NavTabs store={this.props.store} token={token}/>
                                <div className="tab-content">
                                    <div className="tab-pane show active">
                                        {this.setTab()}
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                
                <Navbar token={token}/>
            
            </div>
        );
    }
}

export default HomePage;