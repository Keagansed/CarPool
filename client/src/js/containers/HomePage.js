// File Type: Component

import { observer } from "mobx-react";
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import AddIcon from '@material-ui/icons/Add';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import SettingsIcon from '@material-ui/icons/Settings';
import HomeIcon from '@material-ui/icons/Home';
import UserIcon from '@material-ui/icons/Person';
import { Link } from 'react-router-dom';

import Carpools from '../components/carpool/Carpools';
import NewRoute from '../components/route/NewRoute';
import OffersStore from '../stores/OffersStore';
import Routes from '../components/route/Routes';
import RoutesStore from '../stores/RoutesStore';
import Trips from '../components/trip/Trips';
import VerifyWrapper from './VerifyWrapper';


import { getFromStorage } from '../utils/localStorage.js';

//Container
function TabContainer({ children, dir }) {
    return (
        <Typography component="div" dir={dir} style={{ marginTop: 48, marginBottom: 56 }}>
            {children}
        </Typography>
    );
}
TabContainer.propTypes = {
    children: PropTypes.node.isRequired,
};

//Styling specific to this page
const styles = theme => ({
    root: {
        flexGrow: 1,
        backgroundColor: theme.palette.background.paper,
    },
    topNav: {
        position: 'fixed',
        top: 0,
    },
    bottomNav: {
        width: '100%',
        position: 'fixed',
        bottom: 0,
        borderTop: '1px solid lightgrey',
        backgroundColor: '#3f51b5',
    },
});

/*
* Purpose: Container compenent for the application HomePage, only takes care of tab switching all data
* is in sub-components
*/
@observer class HomePage extends Component {
    constructor() {
        super()

        this.state = {
            token: "",
        }
    }

    /*
    * Purpose: Verifies the users token before component mounting to make sure the user is authorised
    */
    componentWillMount() {
        const obj = getFromStorage('sessionKey');
        let { token } = obj;

        this.setState({
            token,
        })

        let { store } = this.props;
        store.token = token;
    }

    //Handle tab changes
    handleChange = (event, value) => {
        let { store } = this.props;
        store.setTab(value);
    };

    render() {
        const { classes } = this.props;
        const { token } = this.state;
        const { activeTab } = this.props.store;

        return (
                <div className={classes.root}>
                    <AppBar className={classes.topNav}>
                        <Tabs value={activeTab} onChange={this.handleChange} fullWidth>
                            <Tab label="Routes" />
                            <Tab label="Carpools" />
                            <Tab label="Trips" />
                            <Tab icon={<AddIcon />} href="#basic-tabs" />
                        </Tabs>
                    </AppBar>
                    {activeTab === 0 && <TabContainer><Routes store={RoutesStore} token={token} /></TabContainer>}
                    {activeTab === 1 && <TabContainer><Carpools store={OffersStore} token={token} /></TabContainer>}
                    {activeTab === 2 && <TabContainer><Trips /></TabContainer>}
                    {activeTab === 3 && <TabContainer><NewRoute store={RoutesStore} token={this.props.store.token} /></TabContainer>}
                    <BottomNavigation
                        value={1}
                        className={classes.bottomNav}
                    >
                        <BottomNavigationAction 
                            component={Link}
                            to={{pathname: "/ProfilePage/" + token}}
                            label="Profile" 
                            icon={<UserIcon />} 
                            showLabel 
                        />
                        <BottomNavigationAction 
                            label="Home" 
                            icon={<HomeIcon />} 
                            showLabel 
                            style={{color: 'white'}}
                        />
                        <BottomNavigationAction 
                            component={Link}
                            to={`/Settings`}
                            label="Settings" 
                            icon={<SettingsIcon />} 
                            showLabel
                        />
                    </BottomNavigation>
                </div>
        );
    }
}

HomePage.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(VerifyWrapper(HomePage));