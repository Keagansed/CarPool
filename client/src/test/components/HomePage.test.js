import React from 'react';
import HomePage from '../../js/components/home/HomePage';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';


describe('Home Page Component', () => {
    const mockStore = configureStore();
    let store, container;
    const initialState = {token:"123Foo"};

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(<MemoryRouter><HomePage store={store} /></MemoryRouter>);
    });

    it('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><HomePage store={store} /></MemoryRouter>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });

    it('renders correctly', () => {
        expect(container.length).toEqual(1);
    });
});