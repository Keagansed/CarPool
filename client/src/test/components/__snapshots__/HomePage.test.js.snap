// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Component captures snapshot 1`] = `
<div
  className="HomePage"
>
  <div
    className="p-0 page-content"
  >
    <div
      className="container-fluid m-0"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 p-0"
        >
          <ul
            className="nav nav-tabs bg-primary fixed"
          >
            <li
              className="nav-item w-25 bg-info text-center px-0"
            >
              <a
                className="nav-link bg-info active"
                href=""
                onClick={[Function]}
              >
                <b>
                  <b>
                    Routes
                  </b>
                </b>
              </a>
            </li>
            <li
              className="nav-item w-25 text-center bg-info"
            >
              <a
                className="nav-link bg-info "
                href=""
                onClick={[Function]}
              >
                <b>
                  <b>
                    Carpools
                  </b>
                </b>
              </a>
            </li>
            <li
              className="nav-item w-25 text-center bg-info"
            >
              <a
                className="nav-link bg-info "
                href=""
                onClick={[Function]}
              >
                <b>
                  <b>
                    Trips
                  </b>
                </b>
              </a>
            </li>
            <li
              className="nav-item w-25 bg-info text-center flex-center no-active"
            >
              <div
                className="nav-link p-0 bg-info no-active"
                href=""
              >
                <a
                  className="text-secondary"
                  href="/newRoute"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-plus fa-lg fa-fw d-inline nav-item-circle"
                    onClick={undefined}
                  />
                </a>
              </div>
            </li>
          </ul>
          <div
            className="tab-content"
          >
            <div
              className="tab-pane show active"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="navbar-component navbarBottom"
  >
    <div
      className="container-fluid p-0"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 "
        >
          <ul
            className="nav nav-pills navbarBottom text-center"
          >
            <li
              className="nav-item ml-auto w-25"
            >
              <a
                className="nav-link no-hover"
                href="/Setting"
                onClick={[Function]}
              >
                <i
                  className="fa fa-cog fa-2x text-secondary d-inline"
                />
                Â 
              </a>
            </li>
            <li
              className="nav-item w-25"
            >
              <a
                className="nav-link no-hover"
                href="/HomePage"
                onClick={[Function]}
              >
                <i
                  className="fa fa-fw fa-car fa-2x d-inline text-secondary"
                />
              </a>
            </li>
            <li
              className="nav-item mr-auto text-secondary w-25"
            >
              <a
                className="nav-link no-hover"
                href="/ProfilePage/undefined"
                onClick={[Function]}
              >
                <i
                  className="fa fa-fw fa-2x fa-user d-inline text-secondary"
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
